.github/workflows/lint.yaml

```diff
@@ -40,10 +43,24 @@ jobs:
         with:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Install LLVM ${{ matrix.llvm }}
+        run: |
+          if ((Get-WinGetPackage -Id LLVM -Source winget -MatchOption Equals).InstalledVersion -eq '${{ matrix.llvm }}') {
+            Write-Host "LLVM ${{ matrix.llvm }} is already installed."
+          } else {
+            Write-Host "Installing LLVM ${{ matrix.llvm }}..."
+            Install-WinGetPackage -Id LLVM.LLVM -Version ${{ matrix.llvm }} -Source winget -MatchOption Equals -Mode Silent -Force
+          }
+          clang --version
+
       - name: Install WDK (${{ matrix.wdk }})
         run: |
-          if (!(winget list --exact --source winget --id ${{ matrix.wdk }})[-1].contains("${{ matrix.wdk }}")) {
-            winget install --disable-interactivity --source winget --exact --id ${{ matrix.wdk }}
+          if ((Get-WinGetPackage -Id ${{ matrix.wdk }} -Source winget -MatchOption Equals).Id -eq '${{ matrix.wdk }}') {
```

Comment:
Can repeated commands be shared between workflows? I imagine this should be possible

Reply:
These pwsh tasks could be turned into composite github actions, and then grouping of actions could also be done. Full workflows could also be reused between this the WDR repo and the WDR samples repo, but currently the workflows are slightly different.